assets/js/ViewerApp.jsx

```diff
@@ -22,26 +24,46 @@ class ViewerApp extends Component {
       .join()
       .receive('ok', () => {});
 
-    channel.on('sign_update', ({ sign_id: signId, line_number: lineNumber, text: content }) => {
+    channel.on('sign_update', ({
+      sign_id: signId, duration, line_number: lineNumber, text: content,
+    }) => {
       this.setState(prevState => ({
-        signs: updateSigns(prevState.signs, { signId, lineNumber, content }),
+        signs: updateSigns(prevState.signs, {
+          signId, duration, lineNumber, content,
+        }),
       }));
     });
+
+    this.timerID = setInterval(
+      () => this.updateTime(),
+      1000,
+    );
   }
 
-  render() {
-    const { signs } = this.state;
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+  }
 
+  updateTime() {
+    this.setState(({
```

Comment:
nitpick: I think this gained an extra set of () when it was previously a function (which you don't need to pass it as in this case)

Reply:
whoops